kernel/kernel-api/src/test/scala/io/delta/kernel/internal/util/PartitionUtilsSuite.scala

```diff
@@ -174,12 +232,43 @@ class PartitionUtilsSuite extends AnyFunSuite {
       }
   }
 
-  private def col(names: String*): Column = {
-    new Column(names.toArray)
+  val defaultExprHandler = new ExpressionHandler {
+    override def getEvaluator(inputSchema: StructType, expression: Expression, outputType: DataType)
+    : ExpressionEvaluator =
+      throw new UnsupportedOperationException("Not implemented")
+
+    override def getPredicateEvaluator(inputSchema: StructType, predicate: Predicate)
+    : PredicateEvaluator = {
+      def hasSupportedExpr(expr: Expression): Boolean = {
+        expr match {
+          case _: Column | _: Literal => true
+          case pred: Predicate =>
+            pred.getName.toUpperCase() match {
+              case "AND" | "OR" | "=" | "!=" | ">" | ">=" | "<" | "<=" =>
+                !pred.getChildren.asScala.exists(!hasSupportedExpr(_))
+              case _ => false
+            }
+          case _ => false
+        }
+      }
+      if (hasSupportedExpr(predicate)) return null
+      throw new UnsupportedOperationException("Has unsupported expression");
+    }
+
+    override def createSelectionVector(values: Array[Boolean], from: Int, to: Int): ColumnVector =
+      throw new UnsupportedOperationException("Not implemented")
   }
 
-  private def predicate(name: String, children: Expression*): Predicate = {
+  private def col(names: String*): Column = new Column(names.toArray)
+  private def predicate(name: String, children: Expression*): Predicate =
     new Predicate(name, children.asJava)
-  }
+  private def and(left: Predicate, right: Predicate): Predicate = predicate("AND", left, right)
```

Comment:
these will be removed once https://github.com/delta-io/delta/pull/2492 is merged